### DISCUSSION TITLE
How do I get the error thrown by the setup method of an SFC?

### DISCUSSION THREAD

**Discussion by trajano** [vuejs/router#1749] [Help and Questions] [answered]
Say I have an SFC that checks if based on the data provided from its props it throws `NotFoundError` if the data is not available.  I am making my SFC not tied to Vue-Router so there's no `useRouter` code in the component.

I tried to use `onError` but it does not appear to catch it.

Using `afterEach` the `failure` parameter just returns undefined.

I'd like to keep the logic with the SFC rather than implementing the logic on the router itself.

**Comment by posva** [comment, marked as answer]
there is `router.onError()` if the error happens within the navigation (e.g. a guard or fetching the component)
